File: solsim-server/models/dtos.go
package models

type SolWebSocketReq struct {
	Settings    CalcSettings `json:"Settings"`
	OrbitBodies []Body       `json:"OrbitBodies"`
}

type SolWebSocketResp struct {
	CalcData    CalcData `json:"CalcData"`
	OrbitBodies []Body   `json:"OrbitBodies"`
}

File: solsim-client/src/app/models/dtos.ts
import { CalculationSettings, OrbitBody } from "./models";

export interface SolWebSocketDataRes {
    CalcData: any;
    OrbitBodies: OrbitBody[];
  }
  
  
  export interface SolWebSocketDataReq {
    Settings: CalculationSettings;
    OrbitBodies: OrbitBody[];
  }